includes/data-stores/class-wc-product-data-store-cpt.php

```diff
@@ -569,6 +569,14 @@ protected function update_post_meta( &$product, $force = false ) {
 				case 'date_on_sale_to':
 					$value = $value ? $value->getTimestamp() : '';
 					break;
+				case 'stock_quantity':
+					// Fire actions to let 3rd parties know the stock is about to be changed.
+					if ( $product->is_type( 'variation' ) ) {
+						do_action( 'woocommerce_variation_before_set_stock', $product );
+					} else {
+						do_action( 'woocommerce_product_before_set_stock', $product );
+					}
```

Comment:
Hi @pjv, the changes look fine but before merging the pull request it would be great to have some documentation for the hooks. Could you please merge in this change?

```suggestion
					if ( $product->is_type( 'variation' ) ) {
						/**
						* Action to signal that the value of 'stock_quantity' for a variation is about to change.
						*
						* @since 4.9
						*
						* @param int $product The variation whose stock is about to change.
						*/
						do_action( 'woocommerce_variation_before_set_stock', $product );
					} else {
						/**
						* Action to signal that the value of 'stock_quantity' for a product is about to change.
						*
						* @since 4.9
						*
						* @param int $product The product whose stock is about to change.
						*/
						do_action( 'woocommerce_product_before_set_stock', $product );
					}
```

Reply:
@Konamiman Thanks for the changes - merged. 

Does the @ since 4.9 mean that I should be expecting to maintain my private branch on my production WC install until you guys release WC version 4.9?